test/sql/alter/alter_col/test_not_null_multi_tran.test

```diff
@@ -0,0 +1,100 @@
+# name: test/sql/alter/not_null/test_not_null_multi_tran.test
+# description: Test Set/Drop NOT NULL in multiple transactions
+# group: [alter_col]
+
+statement ok
+PRAGMA enable_verification
+
+# Scenario #1, tran1 set not null, tran2 not able to insert
+statement ok con1
+CREATE TABLE t(i INTEGER, j INTEGER)
+
+statement ok con1
+BEGIN TRANSACTION
+
+statement ok con1
+ALTER TABLE t ALTER COLUMN j SET NOT NULL
+
+statement ok con2
+BEGIN TRANSACTION
+
+statement error con2
+INSERT INTO t VALUES(7777, NULL)
+
+statement ok con1
+INSERT INTO t VALUES(7, 7)
+
+statement ok con2
+ROLLBACK
+
+statement ok con1
+COMMIT
+
+query I con1
+SELECT i FROM t
+----
+7
+
+# Scenario #2, tran1 set not null, tran2 failed to commit
+statement ok con1
+DROP TABLE IF EXISTS t
+
+statement ok con1
+CREATE TABLE t(i INTEGER, j INTEGER)
+
+statement ok con1
+BEGIN TRANSACTION
+
+statement ok con2
+BEGIN TRANSACTION
+
+statement ok con2
+INSERT INTO t VALUES(7777, NULL)
+
+statement ok con1
+ALTER TABLE t ALTER COLUMN j SET NOT NULL
+
+statement ok con1
+COMMIT
+
+statement error con2
+COMMIT
+
+statement error con1
+INSERT INTO t VALUES(7777, NULL)
+
+query I con1
+SELECT COUNT(*) FROM  t
+----
+0
+
+# Scenario #3, cannot insert if one connection alter column
+statement ok con1
+DROP TABLE IF EXISTS t
+
+statement ok con1
+CREATE TABLE t(i INTEGER, j INTEGER)
+
+statement ok con1
+BEGIN TRANSACTION
+
+statement ok con1
+ALTER TABLE t ALTER COLUMN j SET NOT NULL
+
+statement ok con2
+BEGIN TRANSACTION
+
+statement error con2
+INSERT INTO t VALUES(1, 1)
+
+statement ok con1
+COMMIT
+
+statement ok con2
+COMMIT
+
+query I con1
+SELECT count(*) from t
+----
```

Comment:
Can we add a test where we add a constraint and drop it again in the same transaction?

Reply:
Sorry I don't get your point again, there's a test here for single transaction: [test/sql/alter/alter_col/test_not_null_in_tran.test:L80](https://github.com/duckdb/duckdb/pull/4185/files#diff-e442cf794434022088f348adf059fd68482e532b77a96003f867ceafc1488338R80-R100)
Is that OK?

Reply:
Ah, I missed that one. Thanks for pointing that out. What is missing there is an insert that fails the constraint prior to the drop, and an insert that succeeds after the drop constraint. Could you add those?

Reply:
As a failed insertion would abort the current explicit transaction, I assume you mean test set/insert_fail/drop/insert_ok in auto commit mode. Just added the test. Please let me know if I misunderstood anything.